shpc/main/modules/templates/singularity.tcl

```diff
@@ -74,6 +74,11 @@ set inspectCmd "singularity \${SINGULARITY_OPTS} inspect \${SINGULARITY_COMMAND_
 # set_shell_function takes bashStr and cshStr
 set-alias {|module_name|}-shell "${shellCmd}"
```

Comment:
ah so one interesting thing I just ran into (that I think we should find a workaround for) is for modules to be portable. E.g., right now the module paths are hardcoded, and for this reason I couldn't find the clingo script (I have to run a container to test).

```bash

-- add path with "alias" wrapper scripts to PATH
prepend_path("PATH", "/home/vanessa/Desktop/Code/shpc/modules/ghcr.io/autamus/clingo/5.5.1/bin")
```
What I think we want to do is figure out how to either have a top level config file that sets a variable for the root, OR to calculate on the fly. I haven't found good examples yet (most are for Lua as a script) so i don't have good suggestions. But I do think it would be nicer if we could move an entire module directory without it breaking!

Reply:
This is a really interesting insight, I had never encountered so far this type of scenario, in which software paths in a deployment may change. This is because most often in HPC systems, the software deployment is done and set in stone. 
I think it's something worth tackling!

How about the following ? (just using a meta language to convey the idea):
```
if SHPC_MODULE_ROOT is defined  
    set PATH to "$SHPC_MODULE_ROOT/bla-bla-bla/bin"
otherwise 
    set PATH to "hardcoded path"
```
The 1st case is for when you need the flexibility, which you can achieve by means of an env. variable.
The 2nd case is for the scenario where the deployment is set in stone.

Thoughts?

Reply:
I love the idea, but how do we derive the root? I was poking around LMOD and environment modules docs and couldn't figure it out. Should we ping someone with expertise?

Reply:
In my proposition above, I was thinking SHPC_MODULE_ROOT would be defined manually by the sys admin, for those specific cases that are needed.

I gave some extra thought about this point, and in the end these wrapper scripts are similar in essence to the container SIFs in SHPC (and actually I now realise they should sit in a sub-dir of the container dir, more on this later): you hard code containerPath in the module, and can't change that, and similar story goes for the wrapper scripts.

If it's fine to have the container path hardcoded, I think it should be similarly fine to have the wrapper path hardcoded.

In addition, though, as I said, I realise the wrappers should sit in a `bin/` under the container path; in fact, under the module path they risk slowing module loading (a bit like the containers were doing in that other ticket by some other user of SHPC).
But then, if container and wrapper are in the same root directory... we could really leverage the syntax I was proposing above, in case you need to make things flexible: having a SHPC_CONTAINER_ROOT that is defined when needed, to make finding container/wrappers dynamic at loading time!

thoughts?

Reply:
That’s a good idea! I think what we could do is take any setting and allow it to be set in the environment with SHPC_<NAME> and then have an envar that falls back to the hard coded path.

I do wonder however if there is still some way for the module software to derive it? I haven’t found anything so far but maybe we can ask @xdelaruelle !

Reply:
So, this does the relative path trick for Lmod!
```
prepend_path("PATH", pathJoin(string.gsub(myFileName(), "/module.lua", ""), "bin" ) )
```

Reply:
and for TCL the right module variable should be `ModulesCurrentModulefile`, now just have to write the substitution function in TCL 
(be back later)

Reply:
without hard-coding the module filename (Lmod):
```
prepend_path("PATH", pathJoin(myFileName():match("(.*[/])") or "./", "bin") )
```

Reply:
and here is the Tcl version!

```
prepend-path PATH "[file dirname ${ModulesCurrentModulefile}]/bin"
```

Reply:
these will do, provided the aliases sit in a subdir of where the modulefile is (or in the same dir, for what matters).
will work more on the code tomorrow hopefully

Reply:
You're amazing! I tried looking for both of those and came up short for both. This is super great! :partying_face:

Reply:
...and now it's in the PR, just committed! 🎉